src/numerics.py

```diff
@@ -0,0 +1,37 @@
+import numpy as np
+
+
+class Numerics:
+    def __init__(self, efield, nstate, threshold):
```

Comment:
I suggest that you take the entire data object as input to the constructor and then extract efield, nstate by
efield = data.efield
nstate = data.nstate_i

Otherwise, the signature could become arbitrarily long - theoretically ;)